catt/controllers.py

```diff
@@ -215,32 +219,33 @@ def __init__(self):
 
         if not self.store_path.is_file():
             devices = get_chromecasts()
-            cache_data = {d.name: self._create_device_entry(d.host, d.port) for d in devices}
+            cache_data = {
+                d.name: CCInfo(d.host, d.port, d.device.manufacturer, d.model_name, d.cast_type).all_info
+                for d in devices
+            }
             self._write_store(cache_data)
 
-    def _create_device_entry(self, ip, port):
-        device_data = {"ip": ip}
-        if port != DEFAULT_PORT:
-            device_data["group_port"] = port
-        return device_data
-
     def get_data(self, name: str):  # type: ignore
         data = self._read_store()
         # In the case that cache has been initialized with no cc's on the
         # network, we need to ensure auto-discovery.
         if not data:
-            return (None, None)
+            return None
         if name:
             fetched = data.get(name)
         else:
             # When the user does not specify a device, we need to make an attempt
             # to consistently return the same IP, thus the alphabetical sorting.
             fetched = data[min(data, key=str)]
-        return (fetched["ip"], fetched.get("group_port", 0)) if fetched else (None, None)
+        return (
+            CCInfo(fetched["ip"], fetched["port"], fetched["manufacturer"], fetched["model_name"], fetched["cast_type"])
```

Comment:
Is there a more elegant way of doing this?

Reply:
Yep, `CCInfo(**fetched)`.

Reply:
How can I be sure that the params are fed to `CCInfo` in the right order?

Reply:
They're passed as keyword parameters, so order doesn't matter.

Reply:
Cheers. The relation between positional and keyword parameters is confusing in python. Any best practice you can recommend?

Reply:
Bah I pressed resolve on this by accident, the mobile interface is confusing. Hmm, I tend to call things positionally for small numbers of arguments, if the number of arguments gets larger or they're more convoluted, I tend to specify keywords.

In this case, it's just easier to expand into keywords, rather than specify the parameters one by one.